bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/util/ShellyUtils.java

```diff
@@ -116,24 +121,11 @@ public static void validateRange(String name, Integer value, Integer min, Intege
                 "Value " + name + " is out of range (" + min.toString() + "-" + max.toString() + ")");
     }
 
-    public static String buildSetEventUrl(String localIp, String localPort, String deviceName, Integer index,
-            String deviceType, String urlParm) throws IOException {
-        return SHELLY_URL_SETTINGS + "/" + deviceType + "/" + index + "?" + urlParm + "="
-                + buildCallbackUrl(localIp, localPort, deviceName, index, deviceType, urlParm);
-    }
-
-    public static String buildCallbackUrl(String localIp, String localPort, String deviceName, Integer index,
-            String type, String parameter) throws IOException {
-        String url = "http://" + localIp + ":" + localPort + SHELLY_CALLBACK_URI + "/" + deviceName + "/" + type + "/"
-                + index + "?type=" + StringUtils.substringBefore(parameter, "_url");
-        return urlEncode(url);
-    }
-
-    public static String urlEncode(String input) throws IOException {
+    public static String urlEncode(String input) throws ShellyApiException {
         try {
             return URLEncoder.encode(input, StandardCharsets.UTF_8.toString());
         } catch (UnsupportedEncodingException e) {
-            throw new IOException(
+            throw new ShellyApiException(
                     "Unsupported encoding format: " + StandardCharsets.UTF_8.toString() + ", input=" + input);
```

Comment:
Should you also pass the cause exception?

Reply:
changed